blockchain/chain/src/store/chain_store.rs

```diff
@@ -132,41 +139,51 @@ where
         db: DB,
         chain_config: Arc<ChainConfig>,
         genesis_block_header: &BlockHeader,
+        chain_data_root: &Path,
     ) -> Result<Self>
     where
         DB: Clone,
     {
         let (publisher, _) = broadcast::channel(SINK_CAP);
         let ts_cache = Arc::new(Mutex::new(LruCache::new(DEFAULT_TIPSET_CACHE_SIZE)));
         let genesis_ts = Arc::new(Tipset::from(genesis_block_header));
+        let file_backed_genesis = Mutex::new(FileBacked::new(
+            Some(genesis_block_header.clone()),
+            chain_data_root.join("GENESIS"),
+        ));
+        let file_backed_heaviest_tipset_keys =
+            FileBacked::load_from_file_or_new(chain_data_root.join("HEAD"))?;
```

Comment:
Let's default to using the genesis tipset as the heaviest tipset. This greatly simplifies the rest of the code.

Reply:
Fixed